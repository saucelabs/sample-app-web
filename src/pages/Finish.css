.checkout_complete_container {
  max-width: 1280px;
  margin: 0 auto;
  padding: 30px 15px 50px;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.pony_express {
  width: 100%;
  height: 100%;
  max-width: 523px;
  max-height: 381px;
  margin: 30px auto 50px auto;
}

.complete-header,
.complete-text {
  max-width: 350px;
}

@media only screen and (min-width: 900px) {
  .checkout_complete_container {
    padding-bottom: 100px;
  }
}
