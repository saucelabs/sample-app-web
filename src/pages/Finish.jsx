import React from "react";
import { withRouter } from "react-router-dom";
import PonyExpress from "../assets/img/pony-express.png";
import SwagLabsFooter from "../components/Footer";
import HeaderContainer from "../components/HeaderContainer";
import PropTypes from "prop-types";
import Button, { BUTTON_SIZES } from "../components/Button";
import { ROUTES } from "../utils/Constants";
import "./Finish.css";

const Finish = ({ history }) => {
  return (
    <div id="page_wrapper" className="page_wrapper">
      <div id="contents_wrapper">
        <HeaderContainer secondaryTitle="Checkout: Complete!" />
        <div
          id="checkout_complete_container"
          className="checkout_complete_container"
        >
          <h2 className="complete-header">THANK YOU FOR YOUR ORDER</h2>
          <div className="complete-text">
            Your order has been dispatched, and will arrive just as fast as the
            pony can get there!
          </div>
          <img alt="Pony Express" className="pony_express" src={PonyExpress} />
          <Button
            label="Back Home"
            onClick={() => history.push(ROUTES.INVENTORY)}
            size={BUTTON_SIZES.SMALL}
            testId="back-to-products"
          />
        </div>
      </div>
      <SwagLabsFooter />
    </div>
  );
};
Finish.propTypes = {
  /**
   * The history
   */
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

export default withRouter(Finish);
