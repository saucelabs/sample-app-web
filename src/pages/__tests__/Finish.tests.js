import React from "react";
import { shallow } from "enzyme";
import Finish from "../Finish";

jest.mock("../../utils/shopping-cart");

let props;

describe("CheckOutStepTwo", () => {
  beforeEach(() => {
    props = {
      history: { push: jest.fn() },
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly with default props", () => {
    const wrapper = shallow(<Finish.WrappedComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should redirect when clicking on back home", () => {
    const wrapper = shallow(<Finish.WrappedComponent {...props} />);
    const backHome = wrapper.find("Button").at(0);
    backHome.simulate("click", {
      preventDefault() {},
    });

    expect(props.history.push).toBeCalledWith("/inventory.html");
  });
});
