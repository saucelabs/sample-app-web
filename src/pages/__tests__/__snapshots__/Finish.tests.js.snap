// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckOutStepTwo should render correctly with default props 1`] = `
<div
  className="page_wrapper"
  id="page_wrapper"
>
  <div
    id="contents_wrapper"
  >
    <HeaderContainer
      secondaryTitle="Checkout: Complete!"
    />
    <div
      className="checkout_complete_container"
      id="checkout_complete_container"
    >
      <h2
        className="complete-header"
      >
        THANK YOU FOR YOUR ORDER
      </h2>
      <div
        className="complete-text"
      >
        Your order has been dispatched, and will arrive just as fast as the pony can get there!
      </div>
      <img
        alt="Pony Express"
        className="pony_express"
        src="test-file-stub"
      />
      <Button
        label="Back Home"
        onClick={[Function]}
        size="small"
        testId="back-to-products"
        type="primary"
      />
    </div>
  </div>
  <SwagLabsFooter />
</div>
`;
